tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+        {
+            int index = 0;
+            Model.Symbol[] fieldSymbols = new Model.Symbol[fieldsCollection.Count];
+            foreach (var fieldDefHandle in fieldsCollection)
+            {
+                var fieldDef = MetadataReader.GetFieldDefinition(fieldDefHandle);
+                var fieldTypeName = fieldDef.DecodeSignature(new TypeProvider(false), 0);
+                var fieldName = fieldDef.Name.IsNil ? null : MetadataReader.GetString(fieldDef.Name);
+                fieldSymbols[index] = new Symbol
+                {
+                    Name = fieldName,
+                    Type = fieldTypeName,
+                    SymbolType = ((fieldDef.Attributes & FieldAttributes.Static) != 0) ? SymbolType.StaticField : SymbolType.Field,
+                    Line = UnknownFieldAndArgLine
+                };
+                index++;
+            }
+
+            return fieldSymbols;
+        }
+
+        private void PopulateMethodScopes(TypeDefinition type, MethodDefinitionHandleCollection methods, Model.Scope[] classMethods, ref int index)
+        {
+            foreach (var methodDefHandle in methods)
+            {
+                if (methodDefHandle.IsNil)
+                {
+                    continue;
+                }
+
+                if (DatadogMetadataReader.IsCompilerGeneratedAttributeDefinedOnMethod(methodDefHandle.RowId))
+                {
+                    continue;
+                }
+
+                var methodDef = MetadataReader.GetMethodDefinition(methodDefHandle);
+                var methodScope = CreateMethodScope(type, methodDef);
+                if (methodScope.Scopes != null &&
+                    methodScope is { StartLine: UnknownMethodStartLine, EndLine: UnknownMethodEndLine, SourceFile: null })
+                {
+                    var sourcePdbInfo = GetMethodSourcePdbInfo(methodScope);
+                    methodScope.StartLine = sourcePdbInfo.StartLine;
+                    methodScope.EndLine = sourcePdbInfo.EndLine;
+                    methodScope.SourceFile = sourcePdbInfo.Path;
+                    if (sourcePdbInfo.EndColumn > 0)
+                    {
+                        var ls = new LanguageSpecifics
+                        {
+                            Annotations = methodScope.LanguageSpecifics?.Annotations,
+                            AccessModifiers = methodScope.LanguageSpecifics?.AccessModifiers,
+                            ReturnType = methodScope.LanguageSpecifics?.ReturnType,
+                            StartColumn = sourcePdbInfo.StartColumn,
+                            EndColumn = sourcePdbInfo.EndColumn,
+                        };
+
+                        methodScope.LanguageSpecifics = ls;
+                    }
+                }
+
+                classMethods[index] = methodScope;
+                index++;
+            }
+        }
+
+        private SourcePdbInfo GetMethodSourcePdbInfo(Model.Scope methodScope)
```

Comment:
Should the error handling policy in this potentially be to try/catch and log an error, but ultimately return `null`?

Reply:
Check this [comment](https://github.com/DataDog/dd-trace-dotnet/pull/4782#discussion_r1382376151) please

Reply:
Fixed in d76bac0e9bd3cfbda3e480deff9bbd0e453a32ec